test/support/data_case.ex

```diff
@@ -24,20 +24,43 @@ defmodule Franklin.DataCase do
       import Ecto.Changeset
       import Ecto.Query
       import Franklin.DataCase
+      import Commanded.Assertions.EventAssertions
     end
   end
 
   setup tags do
-    Franklin.DataCase.setup_sandbox(tags)
+    Franklin.DataCase.setup_stores(tags)
+
     :ok
   end
 
   @doc """
-  Sets up the sandbox based on the test tags.
+  Sets up the typical Ecto database sandbox (used for projections) and
+  the event store (which is configured as in-memory for testing only).
   """
-  def setup_sandbox(tags) do
+  def setup_stores(tags) do
+    # Since we stop applications inside of `on_exit` the first thing we want to
+    # do is verify they are all started.
+    {:ok, _apps} = Application.ensure_all_started(:franklin)
+
+    # Next we will start the typical Ecto sandbox like experience which will be
+    # utilized for the projection persistance.
     pid = Ecto.Adapters.SQL.Sandbox.start_owner!(Franklin.Repo, shared: not tags[:async])
-    on_exit(fn -> Ecto.Adapters.SQL.Sandbox.stop_owner(pid) end)
+
+    on_exit(fn ->
+      # Clear the in memory event store.
+      Commanded.EventStore.Adapters.InMemory.reset!(Franklin.CommandedApplication)
+
+      # Stop the Ecto sandbox.
+      Ecto.Adapters.SQL.Sandbox.stop_owner(pid)
+
+      # Stop the app in full. The reason why the app must be stopped and then
+      # started between resetting the event store is to ensure all Commanded
+      # application processes are restarted with their initial state to prevent
+      # state from one test affecting another. FIXME: How does this impact async
+      # tests? I think the InMemory store is global so probably yes. :(
+      :ok = Application.stop(:franklin)
```

Comment:
I think this is a bit of a smell and ideally would like to move the EventStore to something more sandbox-like but for now we'll deal with it.